src/strategies/metis/nettswap/strategy-extraRewarder-WIP.sol

```diff
@@ -0,0 +1,160 @@
+    address public reward = 0x90fE084F877C65e1b577c7b2eA64B8D8dd1AB278;
+    address public masterchef = 0x9d1dbB49b2744A1555EDbF1708D64dC71B0CB052;
+    address[] public extraRewards;
+    address public token0;
+    address public token1;
+
+    // How much REWARD tokens to keep?
+    uint256 public keepREWARD = 1000;
+    uint256 public constant keepREWARDMax = 10000;
+
+    mapping(address => address[]) public swapRoutes;
+
+    uint256 public poolId;
+
+    // **** Getters ****
+    constructor(
+        address _want,
+        uint256 _poolId,
+        address _governance,
+        address _strategist,
+        address _controller,
+        address _timelock
+    )
+        public
+        StrategyBase(_want, _governance, _strategist, _controller, _timelock)
+    {
+        sushiRouter = 0x1E876cCe41B7b844FDe09E38Fa1cf00f213bFf56;
+        IUniswapV2Pair pair = IUniswapV2Pair(_want);
+        token0 = pair.token0();
+        token1 = pair.token1();
+        poolId = _poolId;
+
+        IERC20(token0).approve(sushiRouter, uint256(-1));
+        IERC20(token1).approve(sushiRouter, uint256(-1));
+        IERC20(reward).approve(sushiRouter, uint256(-1));
+    }
+
+    function balanceOfPool() public view override returns (uint256) {
+        (uint256 amount, ) = IRewardChef(masterchef).userInfo(
+            poolId,
+            address(this)
+        );
+        return amount;
+    }
+
+    function getHarvestable() external view returns (uint256) {
+        (uint256 pendingREWARD, , , ) = IRewardChef(masterchef).pendingTokens(
+            poolId,
+            address(this)
+        );
+        return pendingREWARD;
+    }
+
+    // **** Setters ****
+
+    function setKeepREWARD(uint256 _keepREWARD) external {
+        require(msg.sender == timelock, "!timelock");
+        keepREWARD = _keepREWARD;
+    }
+
+    function setRewardTokens(address token) {
+        require(msg.sender == timelock, "!timelock");
+        extraRewards.push(token);
+    }
+
+    function deposit() public override {
+        uint256 _want = IERC20(want).balanceOf(address(this));
+        if (_want > 0) {
+            IERC20(want).safeApprove(masterchef, 0);
+            IERC20(want).safeApprove(masterchef, _want);
+            IRewardChef(masterchef).deposit(poolId, _want);
+        }
+    }
+
+    function _withdrawSome(uint256 _amount)
+        internal
+        override
+        returns (uint256)
+    {
+        IRewardChef(masterchef).withdraw(poolId, _amount);
+        return _amount;
+    }
+
+    function harvest() public override {
+        IRewardChef(masterchef).deposit(poolId, 0);
+        uint256 _reward = IERC20(reward).balanceOf(address(this));
+        uint256 _metis = IERC20(metis).balanceOf(address(this));
+
+        //swap all extraRewards to reward
+        for (uint256 i = 0; i < extraRewards.length; i++) {
+            uint256 _reward = IERC20(extraRewards[i]).balanceOf(address(this));
```

Comment:
This will work if all rewards are continually deposited into the strategy, but generally we will need to claim it from a reward address. To achieve this and make it generic enough for a base strategy, we will need to provide calldata along with each reward token, and an address to detail how to claim those rewards. We can discuss a design for this!